onnxruntime/core/providers/shared/initializer_view/initializer_view.h

```diff
@@ -0,0 +1,52 @@
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#include <memory>
+
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/tensor_shape.h"
+
+#include "gsl/gsl-lite.hpp"
+
+namespace onnxruntime {
+
+class InitializerView {
+ private:
+  InitializerView(const InitializerView&) = delete;
+  InitializerView& operator=(const InitializerView&) = delete;
+  InitializerView() = default;
+
+ public:
+  static std::unique_ptr<InitializerView> Create(
+      const ONNX_NAMESPACE::TensorProto& tensor_proto);
+  const TensorShape& Shape() {
+    return shape_;
+  }
+
+  const DataTypeImpl* Type() {
+    return type_;
+  }
+
+  template <class T>
+  gsl::span<const T> DataAsSpan() {
+    ORT_ENFORCE(unpacked_tensor_.size() >= sizeof(T), " data bytes must more than sizeof (T)");
```

Comment:
a data size of 0 could be valid right?
perhaps we should check that `T` matches the type specified in the TensorProto

Reply:
A shape of {0} is valid for a scalar, but it would still have one element.

Either way, that the contents of the initializer is valid should be checked elsewhere so I'm not sure it's worth the cost of an ORT_ENFORCE here.

Reply:
Seems I can remove this ENFORCE, if unpacked_tensor_.size() < sizepf(T), GSL::span will report failure